src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -106,6 +110,33 @@ AggregatingStep::AggregatingStep(
 {
 }
 
+void AggregatingStep::addConvertingToNullableTransform(QueryPipelineBuilder& pipeline, const BuildQueryPipelineSettings& settings)
+{
+    if (group_by_use_nulls)
+    {
+        auto input_header = pipeline.getHeader();
+        auto dag = std::make_shared<ActionsDAG>(input_header.getColumnsWithTypeAndName());
```

Comment:
Probably using `ActionsDAG::makeConvertingActions` would shorten the code, but the current version is more specific and better for showing intent.